v1/protocol_binding/http_binder.cc

```diff
@@ -0,0 +1,122 @@
+#include "http_binder.h"
+
+namespace cloudevents {
+namespace binding {
+
+using ::io::cloudevents::v1::CloudEvent;
+using ::io::cloudevents::v1::CloudEvent_CloudEventAttribute;
+using ::cloudevents::format::StructuredCloudEvent;
+using ::cloudevents::format::Format;
+using ::cloudevents::format::Formatter;
+using ::cloudevents::formatter_util::FormatterUtil;
+using ::cloudevents::cloudevents_util::CloudEventsUtil;
+
+typedef boost::beast::http::request<boost::beast::http::string_body> HttpRequest;
+typedef absl::flat_hash_map<std::string, CloudEvent_CloudEventAttribute> CeAttrMap;
+
+constexpr absl::string_view kHttpContentKey = "Content-Type";
+
+template <>
+absl::StatusOr<std::string> Binder<HttpRequest>::GetContentType(HttpRequest& http_req) {
+    auto iter = http_req.base().find(kHttpContentKey.data());
+    if (iter == http_req.base().end()) {
+        return std::string("");
+    }
+    return std::string(iter->value());
+}
+
+template <>
+absl::StatusOr<std::string> Binder<HttpRequest>::GetPayload(HttpRequest& http_req) {
+    return http_req.body();
+}
+
+template <>
+absl::StatusOr<CloudEvent> Binder<HttpRequest>::UnbindBinary(HttpRequest& http_req) {
+    CloudEvent cloud_event;
+    for (auto it = http_req.base().begin(); it!=http_req.base().end(); ++it) {
+        std::string key;
+        std::string header_key = (*it).name_string().to_string();
+        if (header_key == kHttpContentKey.data()) {
+            key = kContenttypeKey.data();
+        } 
+        else if (header_key.rfind(kMetadataPrefix.data(), 0) == 0){
+            size_t len_prefix = strlen(kMetadataPrefix.data());
```

Comment:
```
Strlen Returns the length of the C string str.

The length of a C string is determined by the terminating null-character: A C string is as long as the number of characters between the beginning of the string and the terminating null character (without including the terminating null character itself).
```

absl::string_view doesn't guarantee that data will null terminated (implementation defined) therefore you can't expect strlen to work correctly. Please use absl::string_view::size.